server/lib/src/idm/server.rs

```diff
@@ -635,13 +635,29 @@ pub trait IdmServerTransaction<'a> {
                         );
                         return Ok(None);
                     }
-                } else if grace_valid {
-                    security_info!(
-                        "The token grace window is in effect. Assuming parent session valid."
-                    );
                 } else {
-                    security_info!("The token grace window has passed and no entry parent sessions exist. Assuming invalid.");
-                    return Ok(None);
+                    let api_session = entry
+                        .get_ava_as_apitoken_map(Attribute::ApiTokenSession)
+                        .and_then(|sessions| sessions.get(&parent_session_id));
+
+                    if let Some(api_session) = api_session {
+                        if let Some(expiry) = api_session.expiry {
+                            if time::OffsetDateTime::UNIX_EPOCH + ct >= expiry {
```

Comment:
I don't actually think you need to check the time here, since the oauth2 token has it's own expiry that should be *within* the parent sessions expiration window. What you actually need to check is similar to the UAT flow above, where you only need to check presence of the parent session remains. Because of the apiToken is *deleted* then that indicates that the child oauth2 sessions would no longer be valid (global revoke basicly).